### PR TITLE
Elongated Footprint Support

### PR DESCRIPTION
**Pull request by KyleAnthonyShepherd**
Executive Summary:

PR to allow mobile units with a non-square `footprint` to collide and separate like a rectangle, and not as a square dependent on `movedef`. This new feature is explicitly gated behind `modInfo.allowSepAxisCollisionTest = true`. And collision and separation logic for standard square movedef units should be untouched. 

This PR was inspired by:
1. Some drama on the BAR discord, where naval ship design is unnecessarily constrained by the square movedef requirement. 
2. Stress testing Claude Opus 4.6. Vibe coded up to a working state, then a manual review to cut out the cruft and fix edge cases. 

See videos for tests on 3x10 long units (using a 3x3 movedef), and 9x3 wide units (using a 9x9 movedef):

https://github.com/user-attachments/assets/e1e1800f-87b6-4120-a807-1c2733b55712

https://github.com/user-attachments/assets/c6ec3691-7888-4758-bf90-606df6149a5b

One note, the grey model sphere must be large enough to allow units (elongated or not) to push other units out of the way. Related to `CGroundMoveType::CheckCollisionSkid()` I think. 
This is a separate bug? that should be addressed in another PR.

### CODE REVIEW

File: rts/Sim/Units/Unit.h [hunk 1]
```diff
@@ -527,6 +527,17 @@ class CUnit : public CSolidObject
...
+	// gates elongated-footprint collision logic (OBB scan expansion, projected radii, etc.)
+	bool hasElongatedFootprint = false;
+
+	// Pre-computed collision geometry derived from the UnitDef footprint (int2).
+	// Mobile units have xsize/zsize overridden by MoveDef, so these cache the true
+	// physical shape for use in collision detection / OBB tests.
+	// Half-extents in world units: ((footprint - 1) * 0.5 * SQUARE_SIZE) per axis.
+	float2 footprintHalfExtents;
+	// Max half-extent (bounding circle radius): max(footprint.x, footprint.y) - 1) * 0.5 * SQUARE_SIZE.
+	float footprintMaxRadius = 0.0f;
```
Comment (lostsquirrel1) [inline rts/Sim/Units/Unit.h:540]: The footprint changes should be applied to the unit def, not the individual unit. Logically all units of the same type will have the same footprint.

**Review by lostsquirrel1 — COMMENTED** [review]
I feel this would be better handled by an explicit footprintX and footprintZ on the unitDef, which the ground move type can use instead for some arbitrary size calculation (which is what it does at the moment.) In fact, it already has it: they are xsize zsize - so you just need a variable to act as an override and then update groundmove to use it for mobile unit <-> unit collisions.

File: rts/Sim/Units/Unit.h (hunk 1 above)
Comment (sprunk) [reply rts/Sim/Units/Unit.h:540]: Units of the same unitdef type can have a different effective footprint due to differing moveDef via `Spring.MoveCtrl.SetMoveDef` though

Comment (lostsquirrel1) [reply rts/Sim/Units/Unit.h:540]: That is a good point; though you probably still need to be able to calibrate this at unitDef level. Then override at unit level in code if you need.

**Comment by lostsquirrel1** [comment]
With respects to the grey sphere issue: how are you changing that? I believe that should only refer to a unit's collision for projectiles.

**Comment by KyleAnthonyShepherd** [comment]
> With respects to the grey sphere issue: how are you changing that? I believe that should only refer to a unit's collision for projectiles.

I have done no changes to that subsystem in this PR.
I only noticed it when armbats (with a model radius about equal to the short length), was *unable* to push through a crowd of other armbats.

But then when I increased the model radius to the long dimension, it *was* able to push through a crowd of armbats. 

[Also did tests with armthor. Armthors with a small model radius were not pushing other armthors away, but armthors wirh a large model radius easily pushed away other armthors]

File: rts/Sim/Units/Unit.h (hunk 1 above)
Comment (KyleAnthonyShepherd) [reply rts/Sim/Units/Unit.h:540]: footprintX/Z already exist on the unitdef level. But I think the functionality has been basically reduced to the dimensions needed to place the nanoframe on the ground. 

In current master, almost all other unit dimensions are derived from the MoveDef.

So, this PR uses the unitdef footprintX/Z to control collisions and separation IF they are not square.

**Comment by lostsquirrel1** [comment]
That's right, model radius, is correct for determining unit <-> unit collision.

File: rts/Sim/Units/Unit.h (hunk 1 above)
Comment (lostsquirrel1) [reply rts/Sim/Units/Unit.h:540]: Thanks for the clarification. That's fine for mobile unit <-> mobile unit collisions.

There seems to be similar changes for collisions with static units. We can't use elongated collision against buildings because that can interfere with the unit's ability to follow a path given to it by the path finder. We need absolute parity and agreement between the movement and the pathing systems otherwise units can get stuck or behave badly.

mobile units <-> mobile units is fine because the pathing system cannot see mobile units, and mobile units are expected to move around and push each other about.

Comment (KyleAnthonyShepherd) [reply rts/Sim/Units/Unit.h:540]: Yeah, the torture test in the video above, where a 3x10 long unit with a 3x3 pathfinding movedef is forced to turn a corner in a 3-wide hallway, shows that some degree of "phasing/intangability" is needed to prevent stuck situations. 

I suspect there would be a possible construction of a 3x3 pathable passage that a 3x10 would be unable to traverse due to pushout forces.

So some failsafe mechanism to collapse the "collision" volume of a 3x10 unit into its square 3x3 movedef would be needed.
